### PR TITLE
OSD-4552: improve moactl verify quota error messages

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: Ben Ye <yb532204897@gmail.com>

Fixes https://issues.redhat.com/browse/OSD-4552

Readme also updated.

Verification

Screenshot from 2020-07-27 11-44-47

### CODE REVIEW

File: pkg/aws/client.go [hunk 1]
```diff
@@ -335,14 +335,21 @@ func (c *awsClient) GetAccessKeyFromStack(stackName string) (*AccessKey, error)
...
-			return false, fmt.Errorf("Service %s quota code %s %s not valid",
-				quota.ServiceCode, quota.QuotaCode, quota.QuotaName)
+
+		serviceQuota, err := GetServiceQuota(serviceQuotas, quota.QuotaCode)
+		if err != nil {
+			return false, fmt.Errorf("error getting AWS service quota: %s %v", quota.ServiceCode, err)
 		}
+
+		if *serviceQuota.Value < 100 {
+			return false, fmt.Errorf("service %s quota code %s %s not valid, expected quota %f, but got %f",
```
Comment (jeremyeder) [inline pkg/aws/client.go:349]: why not whole integers

Comment (yeya24) [reply pkg/aws/client.go:349]: Thanks for the reivew. I am not very familiar with service quota values, are they all integers? But the value type is float64, I am not sure if it is safe enough to cast to int.

File: pkg/aws/client.go [hunk 2]
```diff
@@ -335,14 +335,21 @@ func (c *awsClient) GetAccessKeyFromStack(stackName string) (*AccessKey, error)
...
-			return false, fmt.Errorf("Service %s quota code %s %s not valid",
-				quota.ServiceCode, quota.QuotaCode, quota.QuotaName)
+
+		serviceQuota, err := GetServiceQuota(serviceQuotas, quota.QuotaCode)
+		if err != nil {
+			return false, fmt.Errorf("error getting AWS service quota: %s %v", quota.ServiceCode, err)
 		}
+
+		if *serviceQuota.Value < *quota.DesiredValue {
+			return false, fmt.Errorf("service %s quota code %s %s not valid, expected quota %f, but got %f",
```
Comment (NautiluX) [inline pkg/aws/client.go:349]:
```suggestion
			return false, fmt.Errorf("Service %s quota code %s %s not valid, expected quota %f, but got %f",
```

File: README.md [hunk 3]
```diff
@@ -161,7 +161,7 @@ $ export AWS_DEFAULT_REGION=us-west-2 && moactl verify quota
 
 I: Validating AWS quota...
 E: Insufficient AWS quotas
-E: Service ebs quota code L-FD252861 Provisioned IOPS SSD (io1) volume storage not valid
+E: service ec2 quota code L-0263D0A3 Number of EIPs - VPC EIPs not valid, expected quota 5.000000, but got 2.000000
```
Comment (NautiluX) [inline README.md:164]:
```suggestion
E: Service ec2 quota code L-0263D0A3 Number of EIPs - VPC EIPs not valid, expected quota 5.000000, but got 2.000000
```

File: pkg/aws/client.go [hunk 4]
```diff
@@ -335,14 +335,21 @@ func (c *awsClient) GetAccessKeyFromStack(stackName string) (*AccessKey, error)
...
+				quota.ServiceCode, quota.QuotaCode, quota.QuotaName, *quota.DesiredValue, *quota.DesiredValue)
```
Comment (NautiluX) [inline pkg/aws/client.go:350]: I assume the last number should be the actual value, not the desired value?

Comment (yeya24) [reply pkg/aws/client.go:350]: You are right! Thanks

File: pkg/aws/client.go (hunk 2 above)
Comment (yeya24) [reply pkg/aws/client.go:349]: It is not good to capitalize error string. So `service` is fine here.

**Comment by yeya24** [comment]
@NautiluX Hello, I addressed all your comments. Please take a look when you get a chance. :)

**Comment by vkareh** [comment]
@yeya24 @NautiluX - I've rebased against master, made the error output into integers, and fixed a few linter issues. Please take a look when you have a chance

**Comment by vkareh** [comment]
just squashed commits. merging

**Pull request merged**